services/flows-operator/src/FlowOperator.js

```diff
@@ -313,6 +340,81 @@ class FlowOperator {
         }, {});
         return Object.assign(envVars, node.env);
     }
+
+    async _createFlowSecret(flow) {
+        this._logger.debug('About to create a secret');
+        const credentials = await this._createAmqpCredentials(flow);
+
+        const flowSecret = new FlowSecret({
+            metadata: {
+                name: flow.id,
+                namespace: this._config.get('NAMESPACE')
+            },
+            data: {
+                AMQP_URI: this._prepareAmqpUri(credentials)
+            }
+        });
+
+        await this._coreClient.secrets.post({
+            body: flowSecret.toDescriptor()
+        });
+
+        this._logger.debug('Secret has been created');
+    }
+
+    async _deleteFlowSecret(flow) {
+        this._logger.debug('About to delete a secret');
+        const secret = await this._getFlowSecret(flow);
+        if (!secret) {
+            return;
+        }
+
+        const username = secret.amqpUsername;
+        await this._deleteAmqpCredentials({ username });
+        this._logger.trace('Removed AMQP credentials');
+
+        await this._coreClient.secrets(secret.id).delete();
+        this._logger.debug('Secret has been deleted');
+    }
+
+    async _getFlowSecret(flow) {
+        try {
+            const result = await this._coreClient.secrets(flow.id).get();
+            return FlowSecret.fromDescriptor(result.body);
+        } catch (e) {
```

Comment:
generally it would be better to handle 404 error and all other errors differently
As 404 is actually ok in our situation. But all other errors are errors, and we can't do nothing with it except to rethrow, let upper layers decide how to handle it.